prebid-server/endpoints/pbs-endpoints-pbc.md

```diff
@@ -122,3 +122,107 @@ Content-Type: application/json
+
+- to cache user-specific data
+- to cache the output of expensive calculations
+- to avoid wide area network calls that cost everyone money
+
+#### Not In-Scope
+
+This feature does not support thread-safe updates of data. The intended use is write-once, read-many. e.g. a module that counted user session depth should not use this feature because that would require many PBS threads to read and atomically update the stored data.
+
+### PBC Configuration
+
+These entries in the PBC application.yaml file define module-related params:
+
+- `api.module-storage-path` - path of module storage POST/GET endpoints. Defaults to '/module-storage'.
+- `api.api-key` - api key to secure module storage POST endpoint. Host companies may not want everyone on the internet to be able to store stuff in their cache. The value of this key is set by the PBS host company in the PBC config and in the PBS config. (See [???](TBD] for details about where to set this.)
+
+Module storage supports different `applications` and storage providers are configured per `application`. For now, only [Redis](https://redis.io/) is supported for module storage. To configure Redis storage for particular application, configure the `module.storage.redis.{application-name}` property.
+If no `applications` are needed, set `module.storage.redis` to `{}` in application.yaml like so:
+
+```yaml
+module:
+  storage:
+    redis: {}
+```
+
+Utilizing all the properties above, sample module storage configuration:
+
+```yaml
+api.module-storage-path: /module-storage
+api.api-key: API_KEY
+module:
+  storage:
+    redis:
+      application:
+        port: 6379
+        host: localhost
+        password: password
+        timeout: 99999
+```
+
+A big difference between module data storage and regular bid/VAST storage is that for module storage, PBC expects the key to be provided in the POST body. This is because the module will need to generate the key from data in the request in order for it to use that same lookup key in future requests.
+
+### POST /module-storage
+
+This endpoint is used to storage module data. 
+
+Endpoint usage is authorized only for the configured api key. PBS or another caller should provide a `x-pbc-api-key` header.
+
+The POST body is JSON. Here is an explanation of the fields:
+
+{: .table .table-bordered .table-striped }
+| Parameter | Scope | Description |
+| --- | --- | --- |
+|	key | required | A name that will be used to reference the stored value. |
+|	value | required | String representation of the data you need to store. |
+|	type | required | Represents the format stored inside the value. Can be one of `JSON`, `XML`, `TEXT`. |
+|	application | required | Configured name of your module storage. Check the configuration section of [Module Storage endpoints](/prebid-server/endpoints/pbs-endpoints-pbc.html#configuration). |
+|	ttlseconds | optional? | How long (in seconds) the data will be available in the module store. Default TBD. |
```

Comment:
I added the "scope" column here -- did I get it right or are all of these required? Is there a default ttlseconds?